dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1339,20 +1339,32 @@ class CompileTest {
     return inDirectory<TaskResult>(testDirectory, () async {
       await flutter('packages', options: <String>['get']);
 
-      final Map<String, dynamic> compileRelease = await _compileApp(reportPackageContentSizes: reportPackageContentSizes);
-      final Map<String, dynamic> compileDebug = await _compileDebug(
+      // "initial" compile required downloading and creating the `android/.gradle` directory while "full"
```

Comment:
Factoring out the invocation that we know triggers a download into its own separate benchmark is a good start, but I think the end state we probably want to shoot for is factoring the download out into a step that the recipe can tag as an infra step rather than a test step.

Reply:
A question is, do we want to track the gradle downloads as part of the build or do we consider it purely setup. The existing tests have the `release_full_compile_millis` shouldering the cost and is technically incorrect compared to the other benchmarks that benefit from not needing to download gradle.

Users will hit this when they try to initially build. Do we thing this extra cost is part of the build cost and thus belonging in the `xxx_full_compile_millis` bench or are we saying one-time initial costs are not a metric we want to track and thus want to separate it out into setup.

Reply:
Hitting the network is not useful to include in a benchmark because we don't have any control over it. Said differently, what would we do if it suddenly became much slower to hit the network?